gateway/sensor.go

```diff
@@ -307,6 +187,85 @@ func (g *Gateway) updateReadings(name string, newReadings map[string]interface{}
 
 	g.lastReadings[name] = readings
 }
+
+func (g *Gateway) DoCommand(ctx context.Context, cmd map[string]interface{}) (map[string]interface{}, error) {
+
+	// Validate that the dependency is correct.
+	if _, ok := cmd["validate"]; ok {
+		return map[string]interface{}{"validate": 1}, nil
+	}
+
+	// Add the nodes to the list of devices.
+	if newNode, ok := cmd["register_device"]; ok {
+		if newN, ok := newNode.(map[string]interface{}); ok {
+			node, err := convertToNode(newN)
+			if err != nil {
+				return nil, err
+			}
+			g.devices[node.NodeName] = node
+		}
+	}
+
+	// Remove a node from the device map and readings map.
+	if name, ok := cmd["remove_device"]; ok {
+		if n, ok := name.(string); ok {
+			delete(g.devices, n)
+			g.readingsMu.Lock()
+			delete(g.lastReadings, n)
+			g.readingsMu.Unlock()
+		}
+
+	}
+
+	return nil, nil
```

Comment:
return empty map[string]interface{} with nill error so that an accessor doesn't panic when trying to use the first return.